content/en/blog/news/2023-kpt-live-apply.md

```diff
@@ -0,0 +1,136 @@
+applications versions with deployments have succeeded. 
+
+This can be seen as  a problem though: the default is that `kpt live apply` tries for 15 minutes before timing out 
+with an error. But since the boot job is normal k8s job the job controller will create new pods on failure up to 
+the backoff limit, that is set to **4**. This can block the application of subsequent changes to the cluster repo.
+
+## Tagging and release notes
+
+When activating `kpt live apply` you will also see that tags are added to the cluster repository upon successful 
+application of changes. These tags can have 3 prefixes:
+
+* `crd` for custom resource definitions
+* `cluster` for cluster wide resources
+* `ns` for namespaced resources (the most common)
+
+For `ns` tags release notes will also be generated and added to the git provider (for example GitHub). This means 
+that you can see what has changed in your cluster by looking at the releases of the cluster repository.
+
+One handy use of this is that you can get a notification from GitHub when changes has been successfully deployed by 
+going to the **Watch** menu for the repo, select **Custom** and check **Releases**. If you get the notification by email or on the
+GitHub site depends on the notification settings for your GitHub account.
+
+This functionality can be further customized by adding scripts to the `extensions` directory of your cluster repository.
+
+* `crd-reconciled` will be executed after a `crd` tag has been added
+* `cluster-reconciled` will be executed after a `cluster` tag has been added
+* `ns-reconciled` will be executed after a `ns` tag has been added
+
+The default behaviour is equivalent to adding the following script as `ns-reconciled`:
+
+```bash
+#!/bin/sh
+
+jx changelog create --tag-prefix ns-
+```
+
+Here is a somewhat hacky variety that sends out a mail with the release notes similarly to what you can 
+subscribe to for yourself:
+
+```bash
+#!/bin/sh
+set -xe
+
+jx changelog create --tag-prefix ns-
+
+token=$(kubectl get secret jx-boot -o jsonpath="{.data['password']}" | base64 -d)
+repourl=$(kubectl get secret jx-boot -o jsonpath="{.data['url']}"    | base64 -d)
+owner="$(echo $repourl | cut -d/ -f4)"
+repo="$(echo $repourl | cut -d/ -f5 | cut -d. -f1)"
+(
+cat <<EOF
+to: changelog@example.com
+From: dev@example.com
+MIME-Version: 1.0
+Return-Path: <>
+Precedence: Bulk
+Content-Type: text/html; charset=utf-8
+Subject: Deploy to kubernetes cluster
+
+<html>
+<body>
+<h1>Deploy to kubernetes cluster</h1>
+EOF
+curl -s  -H "Authorization: Bearer $token" https://api.github.com/graphql -X POST -d '{"query":"query { repository(owner:\"'$owner'\", name:\"'$repo'\") { release(tagName: \"ns-'${TS}'\") { descriptionHTML }}}"}' | yq .data.repository.release.descriptionHTML
+echo "</body></html>"
+) | sendmail -t -S smtp.example.com
+```
+
+To adopt for you own use the minimal changes would be to find and replace the smtp server and mail addresses now 
```

Comment:
To adopt for you own use, the minimal changes you need to do is to find and replace the smtp server and mail addresses (example.com in the code above)

Reply:
I did a slightly different change, I hope you approve. If not I would like to know why.